### PR TITLE
Check public functions for public examples

### PR DESCRIPTION
**Pull request by wilzbach**
To improve Phobos's documentation, here's another addition to the `style` Makefile target.
Here's the idea that _every public function should have at least one public example_ as stated in the [high-level vision](https://wiki.dlang.org/Vision/2016H2) (btw for all public examples the `publictests` Makefile target we will enforce that it's runnable on dlang.org) 

[Script is at the tools repo](https://github.com/dlang/tools/blob/master/styles/has_public_example.d).

Note: this just adds the tool and comes with quite a long blacklist. While I started to fix some modules (e.g. https://github.com/dlang/phobos/pull/4973 and trivial ones in this PR), it's quite a lot of effort. However, once this has been enabled, everyone should be able to help to reduce the blacklist :)

### CODE REVIEW

**Comment by wilzbach** [comment]
Rebased to `upstream/master`, but it seems that CircleCi is experiencing some issues atm:

> Partial System Outage - components affected: GitHub, Heroku, Atlassian Bitbucket Webhooks, Atlassian Bitbucket Website
Updated 3 min ago - Issues with S3 and Job Starts: We have identified the issue with our upstream providers and are monitoring the situation

**Comment by andralex** [comment]
Cool! This is getting close to https://issues.dlang.org/show_bug.cgi?id=16614 as well.

File: std/digest/digest.d [hunk 1]
```diff
@@ -840,7 +840,7 @@ ref T[N] asArray(size_t N, T)(ref T[] source, string errorMsg = "")
      return *cast(T[N]*) source.ptr;
 }
 
-/**
+/*
  * This helper is used internally in the WrapperDigest template, but it might be
  * useful for other purposes as well. It returns the length (in bytes) of the hash value
  * produced by T.
```
Comment (wilzbach) [inline std/digest/digest.d]: That change might be controversial, but there weren't any tests for `digestLength` and the documentation starts with "this helper is used internally", so this change wouldn't expose it on dlang.org.
What's your opinion on this @jpf91? Should we better reword the documentation and add tests?

Comment (jpf91) [inline std/digest/digest.d]: >Should we better reword the documentation and add tests?

Yes, I think that's a better solution.

**Comment by JackStouffer** [comment]
Mind the style failures

**Comment by wilzbach** [comment]
> Mind the style failures

-> https://github.com/dlang/phobos/pull/5215

(though there's one in here as well)

File: std/digest/digest.d (hunk 1 above)
Comment (wilzbach) [inline std/digest/digest.d]: > Yes, I think that's a better solution.

Okay, thanks!
I removed the bit about "used internally only" and added a simple `unittest`.

**Pull request merged**